### ISSUE TITLE
PlayerCoreFactory.xml

### ISSUE THREAD

**Issue by 530330** [koying/SPMC#751]
For me (evolveo H8,AArch64.Android 6.1,API 23) are played well DD and DTS until 5.1, instead of 7.1 are played PCM 2.0.
Archos plays very well DD+ and MXPLayer plays well DDTHD.
I have maked xml
```
<playercorefactory>
        <players>           
                <player name="MXPlayerPro" type="ExternalPlayer" audio="false" video="true">
                        <filename>com.mxtech.videoplayer.pro</filename>
                        <hidexbmc>true</hidexbmc>
                        <playcountminimumtime>120</playcountminimumtime>
						<playonestackitem>true</playonestackitem>
                </player>
                <player name="ArchosVideo" type="ExternalPlayer" audio="false" video="true">
                        <filename>com.archos.mediacenter.video</filename>
                        <hidexbmc>true</hidexbmc>
                        <playcountminimumtime>120</playcountminimumtime>
						<playonestackitem>true</playonestackitem>
                </player>          
		</players>			 
        <rules action="prepend">
				<rule name="DolbyDigital+" audiocodec="eac3" player="ArchosVideo" />
				<rule name="DolbyDigitalTHD" audiocodec="truehd" player="MXPlayerPro" />
        </rules>
</playercorefactory>
```
Unfortunately files are played continuously although I have parameter "Play immediately next file" set off. I cannot stop player.
What is wrong in my script ?

**Comment by cwh060** [comment]
If I had to guess, the rules seem to be set wrong.  

try

```
<rule filename=".*DolbyDigital+.*" audiocodec="eac3" player="ArchosVideo" />
<rule filename=".*DolbyDigitalTHD.*" audiocodec="truehd" player="MXPlayerPro" />
```

Make sure your files have the words DolbyDigital+ & DolbyDigitalTHD in them somewhere. 

I use something similar for my Sony Television with SPMC installed. MXPlayer for most 4K, Sony internal Video player for some 4K and the SPMC DVDplayer for smb playback.  I differentiate the 4k streams by placing a word in the video file name. I'm not sure the Audiocodec=  settings will work. It may not detect eac3 or truehd?  You may have to remove it. I've never had much luck with that or Videoresolution=.

cwh

**Comment by 530330** [comment]
Hm...
I would have to rename files. Unfortunately I can not remember where the codec is, I would have using MediaInfo to examine all files.
Audiocodec= settings really works, but player plays ALL files in folder and no way to stop it. Button STOP or ESC only ended file and continues weiter.

**Comment by cwh060** [comment]
Ok, I played around with a couple files and got ac3 rule to work, and dts to work. 

For dts the codec shows as dts (DTS-HD MA) I had to use `<rule audiocodec="*dts*" player="whateverplayer" />` try using the * symbol. 

As I understand, audiocodec will only pick the first stream, I've not figured out how to pick off another audio stream, but I'm sure there is a way. 

In order to try TrueHD, I will have to slice up a file using ffmpeg. Also, use ffmpeg to find out what the stream names are `ffmpeg -i filename.mkv`
